sys/arduino/include/spiport.hpp

```diff
@@ -0,0 +1,231 @@
+#define SPI_CLOCK_DIV32     (2) /**< Best match for 500 kHz: 400 kHz */
+#define SPI_CLOCK_DIV64     (3) /**< Best match for 250 kHZ: 100 kHz */
+#define SPI_CLOCK_DIV128    (3) /**< Best match for 125 kHz: 100 kHz */
+/** @} */
+
+/**
+ * @name    Arduino compatible bit order values for SPI
+ * @{
+ */
+#define MSBFIRST        (0)     /**< most significat bit first */
+/** @} */
+
+/**
+ * @brief   Arduino SPI configuration interface
+ */
+class SPISettings {
+private:
+    spi_mode_t mode;
+    spi_clk_t clock;
+
+public:
+    /**
+     * @brief   Create a new SPI settings instance
+     *
+     * @param   clock_hz    SPI clock in Hz to use
+     * @param   bitOrder    Has to be `MSBFIST`, for compatibility only
+     * @param   dataMode    SPI mode to use
+     *
+     * The RIOT SPI clock frequency best matching @p clock will be chosen,
+     * but the frequency will never by greater than what is given in @p clock,
+     * unless @p clock_hz is is lower than 100kHz, which is the lowest clock
+     * frequency that will be used.
+     */
+    SPISettings(uint32_t clock_hz, uint8_t bitOrder, uint8_t dataMode);
+
+    /**
+     * @brief   Create a new SPI settings instance with default settings
+     */
+    SPISettings() : SPISettings(4000000, MSBFIRST, SPI_MODE0) { }
+
+    friend class SPIClass;
+};
+
+/**
+ * @brief   Arduino SPI interface
+ *
+ * @warning Wrap all SPI transfers in `SPI.beginTransaction()` and `SPI.endTransaction()`
+ *
+ * The official Arduino SPI-API allows to use SPI transfers without having to
+ * call `SPI.beginTransaction()`, but discourages to do so. The RIOT API does
+ * not provide this feature, instead a call to `SPI.beginTransaction()` is
+ * mandatory. However, most Arduino code already does this and the remaining
+ * code should be fixed anyway.
+ */
+class SPIClass {
+private:
+    spi_t spi_dev;
+    SPISettings settings;
+    bool is_transaction;
+    rmutex_t mut;
+
+public:
+    /**
+     * @brief   Create a new SPI interface instance
+     * @param   spi_dev     The RIOT SPI device to use under the hood
+     */
+    explicit SPIClass(spi_t spi_dev);
+
+    /**
+     * @brief   Create a new SPI interface instance for SPI device 0
+     * @param   uc_pinMISO  Ignored, for compatibility only
+     * @param   uc_pinSCK   Ignored, for compatibility only
+     * @param   uc_pinMOSI  Ignored, for compatibility only
+     * @param   uc_pinSS    Ignored, for compatibility only
+     * @param   uc_mux      Ignored, for compatibility only
+     */
+    SPIClass(uint8_t uc_pinMISO, uint8_t uc_pinSCK, uint8_t uc_pinMOSI,
+             uint8_t uc_pinSS, uint8_t uc_mux) : SPIClass(SPI_DEV(0))
+    {
+        (void)uc_pinMISO;
+        (void)uc_pinSCK;
+        (void)uc_pinMOSI;
+        (void)uc_pinSS;
+        (void)uc_mux;
+    }
+
+    /**
+     * @brief   Transfer a single byte of data
+     * @param[in]   data        Byte to send
+     * @return  The received byte
+     */
+    uint8_t transfer(uint8_t data) {
```

Comment:
Not sure, but I would think that there should be a newline before the opening curly brace.

Reply:
Yep. Thanks for noticing!